src/stan/lang/generator/write_resize_var_idx.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef STAN_LANG_GENERATOR_WRITE_RESIZE_VAR_IDX_HPP
+#define STAN_LANG_GENERATOR_WRITE_RESIZE_VAR_IDX_HPP
+
+#include <ostream>
+
+namespace stan {
+  namespace lang {
+
+    /**
+     * Generate the loop indexes for the first n-1 of the specified
+     * number of array dimensions, writing to the specified stream.
+     *
+     * @param[in] num_ar_dims number of array dimensions of variable
+     * @param[in,out] o stream for generating
+     */
+    void write_resize_var_idx(size_t num_ar_dims,
+                              std::ostream& o) {
+      if (num_ar_dims == 1) return;
```

Comment:
This conditional (line 18) should be removed because the loop handles the boundary cases correctly.

The doc should indicate that `num_ar_dims == 0` should not be allowed, or there should be a test for it up front.  The loop won't handle this case because `0 - 1` is a really big number in unsigned arithmetic.